### PR TITLE
Init Desktop App

### PR DESCRIPTION
**Pull request by brianjcarroll**
This PR does the following

* Ports the homepage and bounty feed components
* Ports the Bounty item view and its components
* Ports the signin and register pages

This PR does NOT include
* creating a bounty
* user settings
* actually logging in (there is no connection to log in yet)

### CODE REVIEW

File: index.html
Comment (tvolk131) [inline index.html:14]: Nit: Can we re-indent this for easier git diffing? Also remove the line below the `</body>` and add a line to the very end of the file.

File: package.json
Comment (tvolk131) [inline package.json:32]: Would you mind taking a pass through all of these added dependencies and making sure they are all needed? It looks like `octokit` isn't used anywhere, maybe there are more we can remove.

File: package.json
Comment (tvolk131) [inline package.json:70]: Nit: Can we add the line break back at the end of this file?

File: postcss.config.js
Comment (tvolk131) [inline postcss.config.js:1]: Not sure exactly what this is doing, but just want to confirm that this file is needed. @ChristianChiarulli do you have more context on this?

File: prettier.config.cjs
Comment (tvolk131) [inline prettier.config.cjs:1]: Does this comment need to end with `**/` rather than `*/`? Also let's clean up the commented-out lines below.

**Review by tvolk131 — COMMENTED** [review]
There's waaaay too much code in this PR for me to review. I looked at most of the configuration code but the rest looks like a direct copy/paste from the other repo. There's a lot of commented-out or otherwise half-finished stuff in the React code. Do we want to comb through it and clean it up before submitting or just slam it through and clean it up later? I'd prefer to keep this repo clean, but I'll leave it up to you guys since I probably won't be touching the React code very much.

I will note that, from testing this, some key features are broken:

* The user profiles route appears to be missing, so clicking on anyone's avatar leads to a blank page where you are stuck unless you close and relaunch the app.
* Simiar issue with the login page - once you are at that page, there's no way to back out

Also I haven't tried putting in my nSec yet, but what exactly does that do? Is it stored in `localStorage`? If so, is this what we want? @ChristianChiarulli and I are also working on a Nostr desktop key management app, which could allow us to completely remove any need for key management inside of this app. Should we just remove the register/login pages?

**Comment by brianjcarroll** [comment]
> There's waaaay too much code in this PR for me to review. I looked at most of the configuration code but the rest looks like a direct copy/paste from the other repo. There's a lot of commented-out or otherwise half-finished stuff in the React code. Do we want to comb through it and clean it up before submitting or just slam it through and clean it up later? I'd prefer to keep this repo clean, but I'll leave it up to you guys since I probably won't be touching the React code very much.
> 
> I will note that, from testing this, some key features are broken:
> 
>     * The user profiles route appears to be missing, so clicking on anyone's avatar leads to a blank page where you are stuck unless you close and relaunch the app.
> 
>     * Simiar issue with the login page - once you are at that page, there's no way to back out
> 
> 
> Also I haven't tried putting in my nSec yet, but what exactly does that do? Is it stored in `localStorage`? If so, is this what we want? @ChristianChiarulli and I are also working on a Nostr desktop key management app, which could allow us to completely remove any need for key management inside of this app. Should we just remove the register/login pages?

Yeah, this PR is basically just one giant copy paste job of essentially the three features I highlighted in the PR summary. There are a few tweaks to remove any dependencies on next (router, auth, etc). I'd like to keep the TODOs and code comments as is for now, because its supposed to reflect whats in the site repo as much as possible (for React / frontend code I mean)

There will be broken key features because a lot depends on functioning npubs/nsecs and none of that is hooked up yet. This is basically to initialize _something_ so that Chris has a foundation to test the desktop signing app.

The next PR will include user profiles/settings/ and actually logging in with an npub (after we discuss the best way to do that), as well as warranted cleanup, once we understand exactly how the site and desktop app diverge

**Comment by tvolk131** [comment]
Chatted with @ChristianChiarulli, here are our takeaways:

1. Tear out all login logic
2. Remove all usage of `window.localStorage` in storing/retrieving public/private keys
3. Make sure profile pages work

Let's get these done before merging this PR

**Review by ChristianChiarulli — APPROVED** [review]

**Comment by ChristianChiarulli** [comment]
ok most features seem to work, let's get this merged. Next we need back buttons, login logic and we also need to update react nostr at some point to include a loading state for batched profiles.

**Pull request merged**
